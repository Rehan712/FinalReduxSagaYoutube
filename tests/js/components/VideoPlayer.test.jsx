import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { VideoPlayer } from '../../../src/js/components/VideoPlayer';

Enzyme.configure({ adapter: new Adapter() });

describe('Test VideoPlayer Component', () => {
	const selectedVideo = {
		id: {
			videoId: ''
		},
		snippet: {
			title: 'naruto',
			description: ''
		}
	};
	const wrapper = shallow(<VideoPlayer selectedVideo={selectedVideo} />);

	it('test the rendering of the component', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('test the data getting from the props', () => {
		expect(selectedVideo.snippet.title).toBe('naruto');
	});
});
