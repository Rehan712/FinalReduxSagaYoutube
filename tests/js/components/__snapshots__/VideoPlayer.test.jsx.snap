// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test VideoPlayer Component test the rendering of the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPlayer
    selectedVideo={
      Object {
        "id": Object {
          "videoId": "",
        },
        "snippet": Object {
          "description": "",
          "title": "naruto",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="player embed-responsive embed-responsive-16by9"
        >
          <iframe
            allowFullScreen={true}
            className="embed-responsive-item"
            src="https://www.youtube.com/embed/"
            title="Video Player"
          />
        </div>,
        <div
          className="video-details"
        >
          <h4
            className="video-title"
          >
            naruto
          </h4>
          <p
            className="video-description"
          >
            
          </p>
        </div>,
      ],
      "className": "videoPlayer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <iframe
            allowFullScreen={true}
            className="embed-responsive-item"
            src="https://www.youtube.com/embed/"
            title="Video Player"
          />,
          "className": "player embed-responsive embed-responsive-16by9",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "allowFullScreen": true,
            "className": "embed-responsive-item",
            "src": "https://www.youtube.com/embed/",
            "title": "Video Player",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="video-title"
            >
              naruto
            </h4>,
            <p
              className="video-description"
            >
              
            </p>,
          ],
          "className": "video-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "naruto",
              "className": "video-title",
            },
            "ref": null,
            "rendered": "naruto",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "video-description",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="player embed-responsive embed-responsive-16by9"
          >
            <iframe
              allowFullScreen={true}
              className="embed-responsive-item"
              src="https://www.youtube.com/embed/"
              title="Video Player"
            />
          </div>,
          <div
            className="video-details"
          >
            <h4
              className="video-title"
            >
              naruto
            </h4>
            <p
              className="video-description"
            >
              
            </p>
          </div>,
        ],
        "className": "videoPlayer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <iframe
              allowFullScreen={true}
              className="embed-responsive-item"
              src="https://www.youtube.com/embed/"
              title="Video Player"
            />,
            "className": "player embed-responsive embed-responsive-16by9",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "allowFullScreen": true,
              "className": "embed-responsive-item",
              "src": "https://www.youtube.com/embed/",
              "title": "Video Player",
            },
            "ref": null,
            "rendered": null,
            "type": "iframe",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="video-title"
              >
                naruto
              </h4>,
              <p
                className="video-description"
              >
                
              </p>,
            ],
            "className": "video-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "naruto",
                "className": "video-title",
              },
              "ref": null,
              "rendered": "naruto",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "video-description",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
